src/include/optimizer/abstract_optimizer_node_contents.h

```diff
@@ -66,7 +66,13 @@ class AbstractOptimizerNodeContents {
    * equal -- simply checks whether OpType and ExpressionType match
    */
   virtual bool operator==(const AbstractOptimizerNodeContents &r) {
-    return GetOpType() == r.GetOpType() && GetExpType() == r.GetExpType();
+    bool equal = GetOpType() == r.GetOpType() && GetExpType() == r.GetExpType();
```

Comment:
It's been almost a year since I've worked on this, but this method confused me a little bit. Specifically why was it only checking the types and not the actual contents? I forget why but my changes didn't work without this change.